src/main/java/org/sonarsource/sonarlint/ls/SonarLintLanguageServer.java

```diff
@@ -379,64 +383,64 @@ public void didSave(DidSaveTextDocumentParams params) {
     });
   }
 
-  // visible for testing
-  void analyze(URI fileUri, String content, boolean shouldFetchServerIssues) {
+  private void analyzeAsync(URI fileUri, String content, boolean shouldFetchServerIssues) {
     if (!fileUri.getScheme().equalsIgnoreCase("file")) {
       LOG.warn("URI '{}' is not a file, analysis not supported", fileUri);
       return;
     }
+    analysisExecutor.execute(() -> {
```

Comment:
If possible, I think this long lambda should be split in a few more palatable chunks (I'm a bit surprised that it did not get flagged by the SQ analysis).